### PR TITLE
Themes: Product highlights plugin improvements

### PR DESCRIPTION
**Pull request by shawnadelic**
Add option for highlights plugin to hide all out-of-stock products (true
by default).

Refs SHOOP-1969

### CODE REVIEW

File: shoop_tests/front/test_general_template_helpers.py [hunk 1]
```diff
@@ -22,6 +27,23 @@ def test_get_root_categories():
...
+    populate_if_required()
+    context = get_jinja_context()
+    cache.clear()
+    # No products sold
+    assert len(list(general.get_best_selling_products(context, n_products=4))) == 0
+
+    product = get_default_product()
+    supplier = get_default_supplier()
+    shop = get_default_shop()
+    order = create_order_with_product(product, supplier, quantity=1, taxless_base_unit_price=50, shop=shop)
```
Comment (tulimaki) [inline shoop_tests/front/test_general_template_helpers.py]: Order is never used

File: shoop/front/template_helpers/general.py [hunk 2]
```diff
@@ -16,12 +16,34 @@
...
-def _get_list_products(request):
-    return Product.objects.list_visible(
+def _get_list_products(context, n_products, ordering=None, filter_dict=None, in_stock_only=True):
+    request = context["request"]
+    customer = request.customer
+    products_qs = Product.objects.all()
+    if ordering:
+        products_qs = products_qs.order_by(ordering)
+    if filter_dict:
+            products_qs = products_qs.filter(**filter_dict)
```
Comment (tulimaki) [inline shoop/front/template_helpers/general.py]: check the indentation here

**Comment by tulimaki** [comment]
retest this please

File: shoop_tests/front/test_general_template_helpers.py [hunk 3]
```diff
@@ -22,6 +28,106 @@ def test_get_root_categories():
...
+    assert product_3 in general.get_best_selling_products(context, n_products=2)
+
+    create_order_with_product(product_1, supplier, quantity=4, taxless_base_unit_price=10, shop=shop)
+    create_order_with_product(product_2, supplier, quantity=4, taxless_base_unit_price=10, shop=shop)
+    cache.clear()
+    # Third product outsold by first two products
+    assert product_3 not in general.get_best_selling_products(context, n_products=2)
+
+
+
```
Comment (tulimaki) [inline shoop_tests/front/test_general_template_helpers.py]: 1 extra blank line

**Comment by tulimaki** [comment]
Actually what you do here is that you add a option to hide unorderable products (not only unstocked itmes). I think the parameters should also refer to showing only orderable products. Also I think the commit message might be little misleading.

**Comment by tulimaki** [comment]
retest this please

**Comment by shawnadelic** [comment]
Just realized that I didn't make changes to the highlight plugin fields in classic_gray

**Comment by tulimaki** [comment]
Looks good, thanks!

**Pull request merged**